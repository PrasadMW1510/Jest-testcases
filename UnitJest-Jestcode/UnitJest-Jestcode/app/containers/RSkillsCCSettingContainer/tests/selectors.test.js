import { fromJS } from 'immutable';
import makeSelectRSkillsCCSettingContainer, * as Selectors from '../selectors';

describe('selectRSkillsCCSettingContainerDomain', () => {
  let mockedState = null;

  beforeEach(() => {
    const rSkillsCCSettingContainer = fromJS({
      error: null,
      stages: 'mockStages',
      loading: false,
      programSettings: 'mockProgramSettings',
      defaultProgramSettings: {},
      defaultProgramSettingsLoading: false,
    });

    mockedState = fromJS({
      rSkillsCCSettingContainer,
    });
  });

  it('should select programSettings substate', () => {
    expect(Selectors.makeProgramSetting()(mockedState)).toMatchSnapshot();
  });

  it('should select stages substate', () => {
    expect(Selectors.makeSelectTestAssignmentStages()(mockedState)).toMatchSnapshot();
  });

  it('should select loading substate', () => {
    expect(Selectors.makeSelectRSkillsCCSettingContainerLoading()(mockedState)).toMatchSnapshot();
  });

  it('should select rSkillsCCSetting Container Domain', () => {
    expect(makeSelectRSkillsCCSettingContainer()(mockedState)).toMatchSnapshot();
  });

  it('should select the default program settings', () => {
    expect(Selectors.makeDefaultProgramSetting()(mockedState)).toMatchSnapshot();
  });

  it('should select default program settings loading substate', () => {
    expect(
      Selectors.makeSelectRSkillsCCSettingDefaultProgramSettingsLoading()(mockedState)
    ).toMatchSnapshot();
  });
});
