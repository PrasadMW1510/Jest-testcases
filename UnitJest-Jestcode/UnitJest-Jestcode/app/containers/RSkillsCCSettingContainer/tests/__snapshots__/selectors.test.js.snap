// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`selectRSkillsCCSettingContainerDomain should select default program settings loading substate 1`] = `false`;

exports[`selectRSkillsCCSettingContainerDomain should select loading substate 1`] = `false`;

exports[`selectRSkillsCCSettingContainerDomain should select programSettings substate 1`] = `"mockProgramSettings"`;

exports[`selectRSkillsCCSettingContainerDomain should select rSkillsCCSetting Container Domain 1`] = `
Immutable.Map {
  error: null,
  stages: "mockStages",
  loading: false,
  programSettings: "mockProgramSettings",
  defaultProgramSettings: Immutable.Map {
  },
  defaultProgramSettingsLoading: false,
}
`;

exports[`selectRSkillsCCSettingContainerDomain should select stages substate 1`] = `"mockStages"`;

exports[`selectRSkillsCCSettingContainerDomain should select the default program settings 1`] = `
Immutable.Map {
}
`;
